### PR TITLE
CircleCI migration of `build and checks` and Nightly job

### PR DESCRIPTION
**Pull request by ishan-krishna-okta**
## Description:
- **What's changed?** Moves Nightly and `build and checks` jobs to CCI.
- **Is this PR related to a Monolith release?** 

### Resolves:

* [OKTA-712147](https://oktainc.atlassian.net/browse/OKTA-712147)
* [OKTA-712148](https://oktainc.atlassian.net/browse/OKTA-712148)

### CODE REVIEW

File: scripts/yarn-checks-ci.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+set -e
+
+source "${0%/*}/helpers.sh"
```
Comment (emikhasyak-okta) [inline scripts/yarn-checks-ci.sh:4]: Can we use `./helpers.sh` or `./scripts/helpers.sh` here?

File: tests/cypress/integration/quickstarts_spec.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ describe('quickstarts page default selections spec', () => {
   const basePage = new BasePage();
 
   beforeEach(() => {
-    basePage.visit('/documentation/');
```
Comment (emikhasyak-okta) [inline tests/cypress/integration/quickstarts_spec.js:9]: Is this change necessary?

**Review by emikhasyak-okta — COMMENTED** [review]
Overall looks good, I left a couple comments

File: tests/cypress/integration/quickstarts_spec.js (hunk 2 above)
Comment (ishan-krishna-okta) [reply tests/cypress/integration/quickstarts_spec.js:9]: Yes, since we moved away from the `yarn dev` step and instead started spinning up the server directly from the build, this test was failing because in `beforeEach` `/documentation` page was throwing 404 (expected but 404s are not handled well in cypress) but in `yarn dev` the page doesn't throw 404 but instead gives a 200 response. Anyway, the logic here was to just change the page to another page when we run a test in the file, which is achieved with visiting `/` route also.

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,14 +1,99 @@
...
+        - checkout
+        - run:
+            name: Install dependencies
+            command: yarn install --frozen-lockfile
+        - run:
+            name: Build 
+            command: yarn build
+        - run:
+            name: Install libnss3
+            command: sudo apt-get update && sudo apt-get install -y libnss3 && sudo apt-get install ca-certificates fonts-liberation libappindicator3-1 libasound2 libatk-bridge2.0-0 libatk1.0-0 libc6 libcairo2 libcups2 libdbus-1-3 libexpat1 libfontconfig1 libgbm1 libgcc1 libglib2.0-0 libgtk-3-0 libnspr4 libnss3 libpango-1.0-0 libpangocairo-1.0-0 libstdc++6 libx11-6 libx11-xcb1 libxcb1 libxcomposite1 libxcursor1 libxdamage1 libxext6 libxfixes3 libxi6 libxrandr2 libxrender1 libxss1 libxtst6 lsb-release wget xdg-utils
```
[the remark this answers is not in the document]
Comment (ishan-krishna-okta) [reply .circleci/config.yml]: Changed the docker version and now this is no longer needed.

File: scripts/yarn-checks-ci.sh (hunk 1 above)
Comment (ishan-krishna-okta) [reply scripts/yarn-checks-ci.sh]: Used `./scripts/helpers.sh`.

**Review by emikhasyak-okta — APPROVED** [review]
LGTM! 🚢
